webui/react/src/Analytics.ts

```diff
@@ -45,21 +47,22 @@ const getReadyAnalytics = (): SegmentAnalytics.AnalyticsJS | undefined => {
   return undefined;
 };
 
-export const setupAnalytics = (info: DeterminedInfo): void => {
+export const setupAnalytics = async (info: DeterminedInfo): Promise<void> => {
   if (!data.analytics) data.analytics = getReadyAnalytics();
   if (!data.analytics || data.isEnabled) return;
 
   /*
    * Segment key should be 32 characters composed of upper case letters,
    * lower case letters and numbers 0-9.
    */
-  const telemetry = info.telemetry;
-  const isEnabled = telemetry.enabled;
-  const isProperKey = telemetry.segmentKey && /^[a-z0-9]{32}$/i.test(telemetry.segmentKey);
-  if (isEnabled && isProperKey) {
-    data.analytics.load(telemetry.segmentKey || '');
-    data.analytics.identify(info.clusterId);
-    data.isEnabled = true;
+  if (info.isTelemetryEnabled) {
+    const telemetry = await getTelemetry({});
+    const isProperKey = telemetry.segmentKey && /^[a-z0-9]{32}$/i.test(telemetry.segmentKey);
+    if (isProperKey) {
+      data.analytics.load(telemetry.segmentKey || '');
+      data.analytics.identify(info.clusterId);
+      data.isEnabled = true;
```

Comment:
**blocker:** I think these lines need to be wrapped in a `try/catch` block as `getTelemetry` can potentially throw an error.
```
try {
  const telemetry = await getTelemetry({});
  ...
} catch (e) {
  handleError({
    error: e,
    message: e.message,
    publicMessage: `Failed to get telemetry info`,
    silent: true,
    type: ErrorType.Server,
  });
}
```

Reply:
Added, thanks.